api/v1beta1/awsloadbalancercontroller_types.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:validation:Enum=AWSShield;AWSWAFv1;AWSWAFv2
+type AWSAddon string
+
+const (
+	AWSAddonShield AWSAddon = "AWSShield"
+	AWSAddonWAFv1  AWSAddon = "AWSWAFv1"
+	AWSAddonWAFv2  AWSAddon = "AWSWAFv2"
+)
+
+// +kubebuilder:validation:Enum=Auto;Manual
+type SubnetTaggingPolicy string
+
+const (
+
+	// AutoSubnetTaggingPolicy enables automatic subnet tagging.
+	AutoSubnetTaggingPolicy SubnetTaggingPolicy = "Auto"
+
+	// ManualSubnetTaggingPolicy disables automatic subnet tagging.
+	ManualSubnetTaggingPolicy SubnetTaggingPolicy = "Manual"
+)
+
+// AWSLoadBalancerControllerSpec defines the desired state of AWSLoadBalancerController
+type AWSLoadBalancerControllerSpec struct {
+
+	// SubnetTagging describes how resource tagging will be done by the operator.
+	//
+	// When in "Auto", the operator will detect the subnets where the load balancers
+	// will be provisioned and have the required resource tags on them. Whereas when
+	// set to manual, this responsibility lies on the user.
+	//
+	// +kubebuilder:default:=Auto
```

Comment:
What transitions are allowed?  Auto to Manual?  Manual to Auto?

Reply:
I [described in details](https://github.com/openshift/aws-load-balancer-operator/pull/93#discussion_r1154115781) what happens in the transitions. Also I updated the comment to explain the transitions to the user.